omod/src/main/java/org/openmrs/module/coreapps/page/controller/summarydashboard/ProgramEnrollmentsPageController.java

```diff
@@ -0,0 +1,78 @@
+package org.openmrs.module.coreapps.page.controller.summarydashboard;
+
+import org.apache.commons.lang.StringUtils;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.node.ObjectNode;
+import org.openmrs.PatientProgram;
+import org.openmrs.Program;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appframework.context.AppContextModel;
+import org.openmrs.module.appframework.domain.AppDescriptor;
+import org.openmrs.module.appframework.domain.Extension;
+import org.openmrs.module.appframework.service.AppFrameworkService;
+import org.openmrs.module.appui.UiSessionContext;
+import org.openmrs.module.coreapps.CoreAppsConstants;
+import org.openmrs.module.coreapps.CoreAppsProperties;
+import org.openmrs.module.emrapi.EmrApiProperties;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.page.PageModel;
+import org.openmrs.ui.framework.page.Redirect;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class ProgramEnrollmentsPageController {
+
+    public Object controller(@RequestParam("app") AppDescriptor app,
+                             @SpringBean EmrApiProperties emrApiProperties,
+                             @SpringBean("coreAppsProperties") CoreAppsProperties coreAppsProperties,
+                             @SpringBean("appFrameworkService") AppFrameworkService appFrameworkService,
+                             PageModel model,
+                             UiSessionContext sessionContext) {
+
+        if (!Context.hasPrivilege(CoreAppsConstants.PRIVILEGE_SUMMARY_DASHBOARD)) {
+            return new Redirect("coreapps", "noAccess", "");
+        }
+
+        model.addAttribute("app", app);
+
+        AppContextModel contextModel = sessionContext.generateAppContextModel();
+        model.addAttribute("appContextModel", contextModel);
+
+        List<Extension> includeFragments = appFrameworkService.getExtensionsForCurrentUser(app.getId() + ".includeFragments", contextModel);
+        Collections.sort(includeFragments);
+        model.addAttribute("includeFragments", includeFragments);
+
+        List<Extension> firstColumnFragments = appFrameworkService.getExtensionsForCurrentUser(app.getId() + ".firstColumnFragments", contextModel);
+        Collections.sort(firstColumnFragments);
+        model.addAttribute("firstColumnFragments", firstColumnFragments);
```

Comment:
Are we only having a first column on this dashboard, not a second? 

I don't think we necessarily need to support two columns, but,in that case, it  might make sense to rename this something else rather than "firstColumn", though admittedly a great name doesn't spring to mind.

Reply:
This is fine, let's just keep this a "firstColumn" for now.